Ft kd code basics

krzy1976 opened the pull request.

krzy1976 committed:
wysyłam taki plik, który mogę utworzyć, dla testu

krzy1976 committed:
Coś próbuję zrobić, ciężko to idzie, choć rzeczy podstawowe :-( wrzucam to co mam na tę chwilę...
Nie wiedziałem jak do pola city z klasy City, przypisać to co użytkownik będzie wpisywał z klawiatury poprzez Scanner, więc utworzyłem od nowa pewnie niepotrzebnie pola city, state i country.

krzy1976 committed:
Wrzucenie plików do odpowiedniej paczki, zastosowanie setterów, zamiana nazwy na userCity

krzy1976 committed:
Zmiana nazwy klasy z City na Location, dopracowanie danych pobieranych od użytkownika (userCity, userCountry)

krzy1976 committed:
Zmiana nazwy klasy z City na Location, dopracowanie danych pobieranych od użytkownika (userCity, userCountry)

krzy1976 committed:
Dodałem menu użytkownika (menu.java)

krzy1976 committed:
Trochę dopracowałem menu użytkownika.

dehydrogenaza commented on WeatherApp/src/main/java/org/example/ApiQuery.java:
```diff
@@ -0,0 +1,7 @@
+package main.java.org.example;
+
+public class ApiQuery {
```
To miał być interfejs, ale teraz myślę że zrobimy z tego klasę zapieczętowaną. Na razie może zostać to co jest.

krzy1976 replied:
ok

dehydrogenaza commented on WeatherApp/src/main/java/org/example/Location.java:
```diff
@@ -0,0 +1,41 @@
+package main.java.org.example;
+
+public class Location {
```
1. To miało implementować/dziedziczyć po ApiQuery.
2. "Location" jest naturalną nazwą dla kilku różnych rzeczy w naszym programie, dlatego proponuję jednak CityQuery (bo ta klasa służy do tworzenia query wyszukującego lokalizacje po miastach).
3. Obiekty mają być niemutowalne.
Do poprawy.

krzy1976 replied:
ok, poprawiam

dehydrogenaza commented on WeatherApp/src/main/java/org/example/LocationMenu.java:
```diff
@@ -0,0 +1,40 @@
+
+public class LocationMenu {
+
+    public static void main(String[] args) {
```
Nie robimy dwóch mainów w programie :)
Do poprawy.

krzy1976 replied:
ok :-)

dehydrogenaza commented on WeatherApp/src/main/java/org/example/LocationMenu.java:
```diff
@@ -0,0 +1,40 @@
+    public static void main(String[] args) {
+
+        Scanner scanner = new Scanner(System.in);
+        Location userLocation = new Location();
```
Jak pisałem w komentarzu przy klasie, Location (czy jak to tam się będzie ostatecznie nazywało) ma być finalne, więc będziesz musiał refactorować proces tworzenia.

dehydrogenaza commented on WeatherApp/src/main/java/org/example/Menu.java:
```diff
@@ -0,0 +1,46 @@
+
+public class Menu {
+
+    public static void main(String[] args) {
```
Metoda 'main' raczej będzie na wyższym poziomie abstrakcji niż Menu.

krzy1976 replied:
ok, ale na razie może tak zostać ?

dehydrogenaza commented on WeatherApp/src/main/java/org/example/Menu.java:
```diff
@@ -0,0 +1,46 @@
+            switch (userOptions) {
+                case "a":
+                    System.out.println("Wybrano opcję 1.");
+                    System.out.println();
```
Te powtarzalne `println()` można wydzielić poza switcha.

krzy1976 replied:
Nie wiem w tym momencie jak, ale może uda mi się coś wymyśleć ;-)

krzy1976 replied:
A wiem, spoko :-)

krzy1976 replied:
ok

dehydrogenaza commented on WeatherApp/src/main/java/org/example/Menu.java:
```diff
@@ -0,0 +1,46 @@
+    public static void main(String[] args) {
+
+        Scanner scanner = new Scanner(System.in);
+        String userOptions = "a";
```
Dziwna trochę konstrukcja, nie ma potrzeby inicjalizować tej zmiennej konkretną opcją, bo to może wprowadzać w błąd na pierwszy rzut oka.
Zrobiłbym po prostu `userOptions = ""`.

krzy1976 replied:
:-) dziwna ale działająca ;-)) poprawiam :-)

krzy1976 replied:
ok

dehydrogenaza commented on WeatherApp/src/main/java/org/example/LocationMenu.java:
```diff
@@ -0,0 +1,40 @@
+
+import java.util.Scanner;
+
+public class LocationMenu {
```
`LocationMenu` dzieli część kodu z `Menu`, można refactorować żeby wydzielić część wspólną do jakiejś klasy nadrzędnej - ale to ma niższy priorytet.
Na razie OK.

dehydrogenaza reviewed (changes requested):
Kilka poprawek.

krzy1976 committed:
Uwzględniłem uwagi Przemka:
-zmiana obiektów na mutowalne w klasie Location która zmieniła nazwę na CityQuery
-zmieniłem metodę main na metodę location w klasie LocationMenu
-przeniosłem powtarzalne println() poza switcha w klasie Menu
-zmieniłem zainicjowanie zmiennej userOptions z "a" na "" w klasie Menu

krzy1976 committed:
To co mam na tę chwilę, jeszcze bez wpisania metod do switcha w klasie Menu

dehydrogenaza reviewed (approved):
looks good 👍🏻
